backend/service/k8s/deployment_test.go

```diff
@@ -364,3 +365,481 @@ func TestProtoForDeploymentStatus(t *testing.T) {
+	var deploymentTestCases = &appsv1.Deployment{
+		Spec: appsv1.DeploymentSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							ReadinessProbe: &v1.Probe{
+								ProbeHandler: v1.ProbeHandler{
+									TCPSocket: &v1.TCPSocketAction{
+										Port: intstr.IntOrString{
+											IntVal: 8080,
+										},
+										Host: "/",
+									},
+								},
+								InitialDelaySeconds:           6,
+								PeriodSeconds:                 26,
+								TimeoutSeconds:                6,
+								SuccessThreshold:              10,
+								FailureThreshold:              9,
+								TerminationGracePeriodSeconds: &terminationVar,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	t.Run("ReadinessProbeTCPAction", func(t *testing.T) {
+		t.Parallel()
+		deployment := ProtoForDeployment("", deploymentTestCases)
+		assert.Equal(t, deploymentTestCases.Spec.Template.Spec.Containers[0].ReadinessProbe.InitialDelaySeconds, *deployment.DeploymentSpec.Template.Spec.Containers[0].ReadinessProbe.InitialDelaySeconds)
+		assert.Equal(t, deploymentTestCases.Spec.Template.Spec.Containers[0].ReadinessProbe.PeriodSeconds, *deployment.DeploymentSpec.Template.Spec.Containers[0].ReadinessProbe.PeriodSeconds)
+
+		probeDeployment := processObjProbe(deploymentTestCases.Spec.Template.Spec.Containers[0].ReadinessProbe)
+		assert.Equal(t, deploymentTestCases.Spec.Template.Spec.Containers[0].ReadinessProbe.TCPSocket.Host, probeDeployment.GetTcpSocket().Host)
+
+		err := updateContainerProbes(deploymentTestCases, &k8sapiv1.UpdateDeploymentRequest_Fields{})
+		assert.NoError(t, err)
+		err = updateContainerProbes(deploymentTestCases, &k8sapiv1.UpdateDeploymentRequest_Fields{
+			ContainerProbes: []*k8sapiv1.UpdateDeploymentRequest_Fields_ContainerProbes{
+				{
+					ReadinessProbe: &k8sapiv1.Probe{
+						Handler: &k8sapiv1.Probe_TcpSocket{
+							TcpSocket: &k8sapiv1.TCPSocketAction{
+								Host: "/test",
+							},
+						},
+						InitialDelaySeconds: newInt32(20),
+						PeriodSeconds:       newInt32(15),
+					},
+				},
+			},
+		})
+		assert.NoError(t, err)
+	})
+}
+
+func TestProtoForDeploymentSpecWithProbesReadinessGRPC(t *testing.T) {
+	t.Parallel()
+	var terminationVar int64 = 30
+	var portGRPC int32 = 8080
+	var serviceGRPC string = "service"
+	var deploymentTestCases = &appsv1.Deployment{
+		Spec: appsv1.DeploymentSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							ReadinessProbe: &v1.Probe{
+								ProbeHandler: v1.ProbeHandler{
+									GRPC: &v1.GRPCAction{
+										Port:    portGRPC,
+										Service: &serviceGRPC,
+									},
+								},
+								InitialDelaySeconds:           4,
+								PeriodSeconds:                 24,
+								TimeoutSeconds:                2,
+								SuccessThreshold:              4,
+								FailureThreshold:              5,
+								TerminationGracePeriodSeconds: &terminationVar,
```

Comment:
Generally should we be testing that if these values are not set, they are at least defaulted correctly?

Reply:
Yes, in most tests I include some attributes and remove others to validate that some are using the default value

K8s include most of the default attributes